### PR TITLE
Fix for Project Manager being able to use remote repo gems that include python tests

### PR DESCRIPTION
**Pull request by spham-amzn**
## What does this PR do?
When adding a remote repository in the the Project Manager, using a gem from that remote repository will fail to configure if it has any Python Tests defined as a target. In this case, the ROS2 gem includes [python smoke tests](https://github.com/o3de/o3de-extras/blob/development/Gems/ROS2/Code/PythonTests/CMakeLists.txt). However, the SDK build intentionally strips out all tests in its prepackaged gem. Since gems that come from external repos do not go through the SDK build process, they don't get stripped out, but this is addressed by baking in `LY_DISABLE_TEST_MODULES=ON` during the SDK build. There is an issue with Linux this property isn't propagated to the PAL_Linux.cmake properly, thus rendering the `PAL_TRAIT_BUILD_TESTS_SUPPORTED` gate to not exclude the test as intended. So when the project is configured, cmake reports the following error
```
-- Configuring done
CMake Error at /opt/O3DE/23.10.1/cmake/Dependencies.cmake:40 (add_dependencies):
  The dependency target "Legacy::Editor" of target "ROS2.PythonTests" does
  not exist.
Call Stack (most recent call first):
  /opt/O3DE/23.10.1/cmake/LYTestWrappers.cmake:246 (ly_add_dependencies)
  /opt/O3DE/23.10.1/cmake/LYTestWrappers.cmake:339 (ly_add_test)
  /home/o3de/O3DE/Gems/ROS2/2.0.0/Code/PythonTests/CMakeLists.txt:9 (ly_add_pytest)

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

This fix addresses that issue propogating the `PAL_TRAIT_BUILD_TESTS_SUPPORTED` property to the generated SDK for Linux. 

Fixes https://github.com/o3de/o3de/issues/16821 

## How was this PR tested?
Performed the following steps:

* Created a debian package
* Downloaded and installed debian package on a clean ubuntu 22.04 image (with ROS2 and O3DE prereq installed)
* Enable the ROS2 environment
* Launched project manager
* Added the stabilization remote repo ( https://raw.githubusercontent.com/o3de/o3de-extras/stabilization/2310 )
* Created new project based on the ROS template (make sure that the ROS2 and other necessary gems are downloaded and enabled)
* Build the project
* Open the Editor

### CODE REVIEW

**Comment by spham-amzn** [comment]
The impact of this PR not going in will be that the ROS2 gem is not compatible with the o3de-sdk install where you specify a remote repo

File: cmake/install/Findo3de.cmake.in [hunk 1]
```diff
@@ -30,6 +30,16 @@ function(o3de_current_file_path path)
...
 
 
+# If the generation of the SDK layout had LY_DISABLE_TEST_MODULES cache variable set, 
+# then the PAL_TRAIT_BUILD_TESTS_SUPPORTED value needs to be propagated as FALSE
+set(disable_test_modules @LY_DISABLE_TEST_MODULES@)
+if (disable_test_modules)
+    set_property(GLOBAL PROPERTY PAL_TRAIT_BUILD_TESTS_SUPPORTED_DEFAULT FALSE)
+else()
+    set_property(GLOBAL PROPERTY PAL_TRAIT_BUILD_TESTS_SUPPORTED_DEFAULT TRUE)
+endif()
```
Comment (AMZN-alexpete) [inline cmake/install/Findo3de.cmake.in:40]: The code on line 49 below this is also trying to address this issue, should it be removed?

File: cmake/Platform/Linux/PAL_linux.cmake [hunk 2]
```diff
@@ -13,7 +13,15 @@ ly_set(PAL_TRAIT_BUILD_HOST_GUI_TOOLS FALSE)
...
 ly_set(PAL_TRAIT_BUILD_UNIFIED_SUPPORTED TRUE)
-ly_set(PAL_TRAIT_BUILD_TESTS_SUPPORTED TRUE)
+
+get_property(is_test_supported_default_set GLOBAL PROPERTY PAL_TRAIT_BUILD_TESTS_SUPPORTED_DEFAULT SET)
+if (is_test_supported_default_set)
+    get_property(test_supported_default GLOBAL PROPERTY PAL_TRAIT_BUILD_TESTS_SUPPORTED_DEFAULT)
+    ly_set(PAL_TRAIT_BUILD_TESTS_SUPPORTED ${test_supported_default})
+else()
+    ly_set(PAL_TRAIT_BUILD_TESTS_SUPPORTED TRUE)
+endif()
```
Comment (lumberyard-employee-dm) [inline cmake/Platform/Linux/PAL_linux.cmake:23]: We do need to propagate this change to the other platforms as well.

File: cmake/install/Findo3de.cmake.in (hunk 1 above)
Comment (lumberyard-employee-dm) [reply cmake/install/Findo3de.cmake.in:40]: Yeah this code can be moved there and setting of the PAL_TRAIT_BUILD_TESTS_SUPPORTED option to false doesn't below can be removed.

Comment (spham-amzn) [reply cmake/install/Findo3de.cmake.in:40]: whoops, i didn't see that little guy down there....

**Review by AMZN-alexpete — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**